backend/src/BIE.Core/BIE.Core.API/Controllers/DatasetController.cs

```diff
@@ -143,7 +160,81 @@ FROM dbo.Hausumringe_mittelfranken
         }
 
         /// <summary>
-        /// Get a record
+        /// Loads the location data for the given point or polygon.
+        /// </summary>
+        /// <param name="request">Contains the current dataset id and the list of coordinates. 
+        /// In case of a single point a list with a single element.</param>
+        /// <returns>Data for the specified point/polygon as a list of key/values.</returns>
+        [HttpPut("loadLocationData")]
+        [ProducesResponseType(typeof(LocationDataResponse), 200)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(500)]
+        public IActionResult LoadLocationData([FromBody, Required] LocationDataRequest request)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            try
+            {
+                var coordinate = request.Location.FirstOrDefault();
+                if (coordinate == null)
+                {
+                    return BadRequest("Location coordinates are required.");
+                }
+
+                var latitude = coordinate.Latitude;
+                var longitude = coordinate.Longitude;
+                var radius = 10000000; // Define the radius as needed
+
+                DbHelper.CreateDbConnection();
+
+                string command = @"
+            SELECT TOP 5
+                Id, 
+                Location.STArea() AS Area, 
+                geography::Point({0}, {1}, 4326).STDistance(Location) AS Distance
+            FROM 
+                dbo.Hausumringe_mittelfranken
+            WHERE 
+                geography::Point({0}, {1}, 4326).STBuffer({2}).STIntersects(Location) = 1
+            ORDER BY 
+                geography::Point({0}, {1}, 4326).STDistance(Location);";
+
+                string formattedQuery = string.Format(command, latitude, longitude, radius);
+
+                var response = new LocationDataResponse
+                {
+                    CurrentDatasetData = new List<DatasetItem>()
+                };
+
+                foreach (var row in DbHelper.GetData(formattedQuery))
+                {
+                    var area = row["Area"];
+                    var distance = row["Distance"];
+                    var location = row["Location"];
```

Comment:
The query is not returning any location parameter. That why getting exception over here.